File: app/src/main/java/org/uns/mtqmnuns/data/entity/Ayah.kt
package org.uns.mtqmnuns.data.entity

import androidx.room.*

@Entity(
    tableName = "Ayah",
    foreignKeys = [
        ForeignKey (
            entity = Surah::class,
            parentColumns = ["id"], 
            childColumns = ["surahId"],
        )
    ],
    indices = [Index(value = ["surahId"])]
)
data class Ayah (
    @PrimaryKey 
    val id: Int,
    
    @ColumnInfo(name = "surahId")  
    val surahId: Int,

    @ColumnInfo(name = "ayahText") 
    val ayahText: String,

    @ColumnInfo(name = "indoText") 
    val indoText: String,

    @ColumnInfo(name = "readText") 
    val readText: String,
)

File: lib/data/entity/ayah.dart
import 'package:drift/drift.dart';
import 'package:inquran/data/entity/surah.dart';

class Ayah extends Table {
  IntColumn get id => integer()();
  IntColumn get surahId => integer().references(Surah, #id)();
  TextColumn get ayahText => text()();
  TextColumn get indoText => text()();
  TextColumn get readText => text()();
  IntColumn get juz => integer()();
  IntColumn get ayahNumber => integer()();
  TextColumn get audioLink => text()();
  IntColumn get page => integer()();
  BoolColumn get favorite => boolean().withDefault(const Constant(false))();

  @override
  Set<Column<Object>> get primaryKey => {id};
}
